src/Domain/FunctionsOfTime/FunctionOfTimeHelpers.hpp

```diff
@@ -58,15 +59,28 @@ struct StoredInfo<MaxDerivPlusOne, false> {
 void reset_expiration_time(const gsl::not_null<double*> prev_expiration_time,
                            const double next_expiration_time);
 
-/// Returns a StoredInfo corresponding to the closest element in the range of
-/// `StoredInfo.time`s that is less than `t`. The function throws an
-/// error if `t` is less than all `StoredInfo.time`s (unless `t` is just less
-/// than the earliest `StoredInfo.time` by roundoff, in which case it returns
-/// the earliest StoredInfo.)
+/*!
+ * \brief Returns a StoredInfo corresponding to the closest element in the range
+ * of `StoredInfo.time`s that is less than `t`.
+ *
+ * The function throws an error if `t` is less than all `StoredInfo.time`s
+ * (unless `t` is just less than the earliest `StoredInfo.time` by roundoff, in
+ * which case it returns the earliest StoredInfo.)
+ *
+ * This entire operation needs to be threadsafe so an additional parameter
+ * `all_info_size` is passed which is the size of `all_stored_infos` that's
+ * allowed to be used in this operator. This is because another thread may be
+ * adding a new element to `all_stored_infos` as this function is called and we
+ * need to guarantee this always returns the same result.
```

Comment:
I don't think this is why the parameter is needed.  The correct result had better be independent of what any other thread is doing.  This is purely to avoid threading UB.